packages/terra-wallet-provider/lib/TerraWalletProvider.ts

```diff
@@ -198,11 +199,15 @@ export default class TerraWalletProvider extends WalletProvider {
         fee: isProto ? Fee.fromData(JSON.parse(data.fee as any)) : Fee.fromAmino(JSON.parse(data.fee as any))
       }
     } else if (data?.msgs) {
+      const gasPrice = data.fee as any
+      const gasLimit = 800000
```

Comment:
Do we want a static `gasLimit` in Terra?

Reply:
Yeah we do and this is used only for init operation. The reason why we need it is to control the spended amount for fee on Init Swap transaction and to match perfectly with the wallet otherwise they are setting their own and sometimes fee could be different in the explorer than the one displayed in the wallet, here is a related task to this issue: https://linear.app/liquality/issue/LIQ-649/calculate-the-max-fee-so-that-there-is-gas-left-for-the-user-to-pay

Reply:
The caller can set the desired fee by passing `data.fee` (checked in the above `if` statement). That's the way most Terra dApps are working and requesting.

The static `gasLimit` is only used for initiateSwap/claim/refund